document/value.go

```diff
@@ -131,56 +122,52 @@ func NewBoolValue(x bool) Value {
 	}
 }
 
-// NewIntValue encodes x and returns a value whose type depends on the
+// NewIntegerValue encodes x and returns a value whose type depends on the
 // magnitude of x.
-func NewIntValue(x int) Value {
-	return intToValue(int64(x))
-}
-
-// NewInt8Value encodes x and returns a value.
-func NewInt8Value(x int8) Value {
+func NewIntegerValue(x int64) Value {
 	return Value{
-		Type: Int8Value,
-		V:    x,
+		Type: IntegerValue,
+		V:    int64(x),
```

Comment:
The PR description says:
> whose encoding size depends on the size of value it holds

I don't see where this is done?

Reply:
This is done in the MessagePack package, everything about encoding is in the `document/encoding` package